controllers/front/ajax.php

```diff
@@ -44,28 +48,28 @@ public function postProcess()
     private function submitHostedFields()
     {
         try {
-            if (!Order::getOrderByCartId($this->context->cart->id)) {
-                $this->validateOrder();
+            /** @var PaymentFrontController $paymentFrontController */
+            $paymentFrontController = $this->module->getService(PaymentFrontController::class);
+
+            if (Order::getOrderByCartId($this->context->cart->id)) {
+                $this->ajaxDie(json_encode([
+                    'error' => true,
+                    'message' => $this->module->l('Order already exists', self::FILE_NAME),
+                    'url' => $this->getRedirectionToControllerUrl('fail'),
+                ]));
             }
 
-            /** @var SaferPayCardAliasRepository $cardAliasRep */
-            $cardAliasRep = $this->module->getService(SaferPayCardAliasRepository::class);
-
-            $selectedCard = Tools::getValue('selectedCard');
-
-            $alias = $cardAliasRep->getSavedCardAliasFromId($selectedCard);
-
-            /** @var SaferPayInitialize $initializeService */
-            $initializeService = $this->module->getService(SaferPayInitialize::class);
-            $initializeBody = $initializeService->initialize(
+            $initializeResponse = $paymentFrontController->create(
+                $this->context->cart,
                 Tools::getValue('paymentMethod'),
                 (int) Tools::getValue(SaferPayConfig::IS_BUSINESS_LICENCE),
-                $selectedCard,
-                $alias,
-                Tools::getValue('fieldToken')
+                Tools::getValue('selectedCard'),
+                Tools::getValue('fieldToken'),
+                ControllerName::SUCCESS_HOSTED,
+                true
             );
-            $this->createSaferPayOrder($initializeBody);
-            $redirectUrl = $this->getRedirectionUrl($initializeBody);
+
+            $redirectUrl = $paymentFrontController->getRedirectionUrl($initializeResponse);
```

Comment:
Not sure if you use any of the response data anyways, maybe returning the redirection URL from create(..) function was an option, otherwise, seems like a service and not controller function anyways